react/src/containers/Home.js

```diff
@@ -1,44 +1,56 @@
-import React, {useEffect} from 'react';
-import { Jumbotron, Col, Form, Button } from 'react-bootstrap';
+import React, {useEffect, useState} from 'react';
+import { Jumbotron, Col, Button } from 'react-bootstrap';
 import NavigationBar from '../component/NavigationBar';
 import {useDispatch, useSelector} from 'react-redux';
-import {getMovies} from '../store/actions/MovieActions';
-import {moviesSelector} from '../store/selectors/MovieSelector';
-import MovieCard from '../component/MovieCard';
+import {getGenres, getMovies, filterMovies} from '../store/actions/MovieActions';
+import {genreSelector, moviesSelector} from '../store/selectors/MovieSelector';
+import MovieList from './MovieList';
+import Search from '../component/Search';
+import Filter from '../component/Filter';
 
 export default function Home() {
 
   const dispatch = useDispatch();
 
+  const [genre, setGenre] = useState(undefined);
+  const [title, setTitle] = useState(undefined);
+
   useEffect(() => {
       dispatch(getMovies());
+      dispatch(getGenres());
   },[]);
 
   const movies = useSelector(moviesSelector());
+  const genres = useSelector(genreSelector());
+
+  const filterMoviesByGenre = (genreId) => {
+    setGenre(genreId);
+
+    const data = {
+      'genreId': genreId,
+      'title': title
+    }
+
+    dispatch(filterMovies(data));
+  };
+
+  const searchMovieByTitle = (titlePar) => {
+    setTitle(titlePar);
+
+    const data = {
+      'genreId': genre,
```

Comment:
```suggestion
      genreId: genre,
```
ne moras da navodis stringove ovde